WordPress/Classes/Utility/WPUserAgent.m

```diff
@@ -35,17 +35,33 @@ - (void)setupUserAgent
     [[NSUserDefaults standardUserDefaults] registerDefaults:dictionary];
 }
 
-#pragma mark - WordPress User-Agent
+#pragma mark - Default and WordPress User-Agents
+
+- (NSString *)defaultUserAgent
+{
+    // Temporarily unset "UserAgent" from registered user defaults so that we
+    // always get the default value, independently from what's currently set as
+    // User-Agent
+    NSDictionary *originalRegisteredDefaults = [[NSUserDefaults standardUserDefaults] volatileDomainForName:NSRegistrationDomain];
+
+    NSMutableDictionary *tempRegisteredDefaults = [NSMutableDictionary dictionaryWithDictionary:originalRegisteredDefaults];
+    [tempRegisteredDefaults removeObjectForKey:WPUserAgentKeyUserAgent];
+    [[NSUserDefaults standardUserDefaults] registerDefaults:tempRegisteredDefaults];
+    
+    NSString *userAgent = [[[UIWebView alloc] init] stringByEvaluatingJavaScriptFromString:@"navigator.userAgent"];
+    NSAssert(userAgent != nil, @"User agent shouldn't be nil");
+    NSAssert(! [userAgent isEmpty], @"User agent shouldn't be empty");
+    
+    [[NSUserDefaults standardUserDefaults] registerDefaults:originalRegisteredDefaults];
+    
+    return userAgent;
+}
```

Comment:
This method seems very thread unfriendly as it is right now.

If the method was called in two separate threads simultaneously it could change the final UserAgent (thread 1 records user Agent A, sets user agent B, thread 2 records user agent B sets user agent C, thread 1 restores A, thread 2 restores B).

If the user agent was requested by another thread while it's temporarily changed we would have a similar problem.  IMHO since the method is supposed to act as a getter, I'd try to solve this by caching the original user agent somewhere and avoid making any changes each time this method is called.